tracer/build/_build/CodeGenerators/CallTargetsGenerator.cs

```diff
@@ -0,0 +1,743 @@
+                            integrationName = namedArgument.Argument.Value?.ToString();
+                            break;
+                        case nameof(InstrumentAttributeProperties.TypeName):
+                            typeName = namedArgument.Argument.Value?.ToString();
+                            break;
+                        case nameof(InstrumentAttributeProperties.TypeNames):
+                            typeNames = GetStringArray(namedArgument.Argument.Value);
+                            break;
+                        case nameof(InstrumentAttributeProperties.MethodName):
+                            methodName = namedArgument.Argument.Value?.ToString();
+                            break;
+                        case nameof(InstrumentAttributeProperties.ReturnTypeName):
+                            returnTypeName = namedArgument.Argument.Value?.ToString();
+                            break;
+                        case nameof(InstrumentAttributeProperties.MinimumVersion):
+                            minimumVersion = namedArgument.Argument.Value?.ToString();
+                            break;
+                        case nameof(InstrumentAttributeProperties.MaximumVersion):
+                            maximumVersion = namedArgument.Argument.Value?.ToString();
+                            break;
+                        case nameof(InstrumentAttributeProperties.ParameterTypeNames):
+                            parameterTypeNames = GetStringArray(namedArgument.Argument.Value);
+                            break;
+                        case nameof(InstrumentAttributeProperties.CallTargetType):
+                            //callTargetType = (namedArgument.Argument.Value as INamedTypeSymbol)?.ToDisplayString();
+                            break;
+                        case nameof(InstrumentAttributeProperties.CallTargetIntegrationKind):
+                            integrationKind = namedArgument.Argument.Value as int?;
+                            break;
+                        case nameof(InstrumentAttributeProperties.InstrumentationCategory):
+                            instrumentationCategory = (InstrumentationCategory)(namedArgument.Argument.Value as uint?).GetValueOrDefault();
+                            break;
+                        default:
+                            hasMisconfiguredInput = true;
+                            break;
+                    }
+
+                    if (hasMisconfiguredInput)
+                    {
+                        break;
+                    }
+                }
+
+                (ushort Major, ushort Minor, ushort Patch) minVersion = default;
+                (ushort Major, ushort Minor, ushort Patch) maxVersion = default;
+                TryGetVersion(minimumVersion, ushort.MinValue, out minVersion);
+                TryGetVersion(maximumVersion, ushort.MaxValue, out maxVersion);
```

Comment:
We should probably handle the return value of `false` here and throw, pointing to the value and the integration with the problem, otherwise we could end up with strange issues:

```suggestion
                if (!TryGetVersion(minimumVersion, ushort.MinValue, out minVersion))
                {
                    throw new InvalidOperationException($"Error: Integration type  '{type}' has invalid value for minimum: '{minimumVersion}'"); 
                }

                if (!TryGetVersion(maximumVersion, ushort.MaxValue, out maxVersion))
                {
                    throw new InvalidOperationException($"Error: Integration type  '{type}' has invalid value for maximum: '{maximumVersion}'"); 
                }
```

Reply:
Agree, but this is already covered in the SourceCodeGenerator. Although I could add more checks here

Reply:
But we don't _want_ the source code generator any more after this PR do we? 🤔

Reply:
Nop, but yep, but nop, but yep... The `IntegrationOptions` is calling `GetIntegrationId`, so... We might want to look for an alternative method in a following PR